crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -217,6 +217,15 @@ def step_allowed?(step)
       false
     end
 
+    # The only case when current step is not the step that should be executed
+    # is when it is already running. In that case, return the next step.
+    def next_step_to_execute
+      step = current_step
+      return step unless running? step
+      i = upgrade_steps_6_7.index step
+      upgrade_steps_6_7[i + 1] || upgrade_steps_6_7.last
```

Comment:
It actually cannot happen. Last step (`:finished` does not move further when it becomes `current_step`, so this method exits on line 224.